tests/db_fixture/database_fixture.cpp

```diff
@@ -671,13 +671,41 @@ fc::variant json_rpc_database_fixture::get_answer( std::string& request )
    return fc::json::from_string( body );
 }
 
-void json_rpc_database_fixture::review_answer( fc::variant& answer, int64_t code, bool is_warning, bool is_fail )
+void check_id_equal( const fc::variant& id_a, const fc::variant& id_b )
+{
+   BOOST_REQUIRE( id_a.get_type() == id_b.get_type() );
+
+   switch( id_a.get_type() )
+   {
+      case fc::variant::int64_type:
+         BOOST_REQUIRE( id_a.as_int64() == id_b.as_int64() );
+         break;
+      case fc::variant::uint64_type:
+         BOOST_REQUIRE( id_a.as_uint64() == id_b.as_uint64() );
+         break;
+      case fc::variant::string_type:
+         BOOST_REQUIRE( id_a.as_string() == id_b.as_string() );
+         break;
+      case fc::variant::null_type:
+         break;
+      default:
+         BOOST_REQUIRE( false );
+   }
+}
+
+void json_rpc_database_fixture::review_answer( fc::variant& answer, int64_t code, bool is_warning, bool is_fail, fc::optional< fc::variant > id )
 {
    fc::variant_object error;
    int64_t answer_code;
 
    if( is_fail )
    {
+      if( id.valid() && code != JSON_RPC_SERVER_ERROR && code != JSON_RPC_INVALID_PARAMS )
+      {
+         BOOST_REQUIRE( answer.get_object().contains( "id" ) );
+         check_id_equal( answer[ "id" ], *id );
+      }
+
```

Comment:
It looks like the new code will forward the ID when these values of `code` occur, so there are at least some cases where we do want tests to enforce that the ID is forwarded when these errors occur.  Let's just require the caller to set `id` to empty if we want to skip this check in a specific test.

Reply:
We can't. Setting the ID to empty is against the jsonrpc standard.